base/remote/tests/testthat/test-fqdn.R

```diff
@@ -1,3 +1,14 @@
 test_that("fqdn() returns exactly one result", {
   expect_length(fqdn(), 1)
 })
+
+test_that("`fqdn()` returns expected `FQDN` value", {
+  
+  #set FQDN
+  Sys.setenv(FQDN = "pecan_host")
+  expect_equal(fqdn(), "pecan_host")
+  
+  #unset FQDN
+  Sys.unsetenv("FQDN")
```

Comment:
Note that this can have persistent side effects after the test is finished (in this case: If another test downstream is expecting to find `$FQDN` set, this will affect it). Wrapping these tests in [withr::with_envvar()](https://withr.r-lib.org/reference/with_envvar.html) would be a clean alternative.